JS/services/utilitiesService.js

```diff
@@ -2,67 +2,101 @@
     var app = angular.module('utilitiesService', []);
```

Comment:
This refactoring of utilities is good stuff. I especially like that it's for making the code more testable.